flake.nix

```diff
@@ -54,6 +61,10 @@
               homepage = "https://git.peppe.rs/languages/statix/about";
               license = licenses.mit;
             };
+
+            checkPhase = ''
+              cargo clippy
+            '';
```

Comment:
Won't this result in the default check phase (I would imagine would be `cargo test`) not being run?

Reply:
good catch.. the checkPhase runs cargo test ... lets move cargo clippy to the preCheckPhase

Reply:
added another commit that now calls `cargo test` and `cargo clippy`. Didn't find a working way to only append the checkPhase.